design/one-pager-template-stacks.md

```diff
@@ -0,0 +1,344 @@
+
+While it may be possible to draw on values from resources outside of the CRD whose template is being defined, we will not entertain that possibility. But if we were to entertain it, we might take advantage of a template function capable of looking up and returning the specified resource.
+
+<!-- TODO(displague) Are there reasons we should not use sub-resource Status or sibling Status values as template variables? -->
+
+## Template Stack Manager
+
+Crossplane `Stack` and `StackInstall` resources are currently managed by the Stack Manager. This controller will reconcile new `StackInstall` resources by examining the metadata included within the requested Stack's filesystem and creating a `Stack` resource that includes `Deployment` to act as a controller for CRDs managed by that Stack. This `Deployment` is defined in the Stack's `spec.controller.deployment` field.
+
+### Deployment
+
+Template Stacks can take advantage of the `spec.controller.deployment` field too. Rather than using a unique controller defined by each Template Stack, the Stack Manager can define a singular common controller for Template Stack resources.
+
+This deployed controller will be responsible for handling **all of the resources** that the Template Stack "owns". It will need a way to identify all of the CRDs and templates the Stack defines. Each GVR could be supplied to the TSM as arguments, but the templates would still need to be resolved. The controller should therefor take a reference to the Stack which contains all of the necessary information for the TSM.
+
+```yaml
+kind: Stack
+spec:
+  controller:
+    deployment:
+      <typical Deployment fields>
+      …
+      containers:
+      - name: template-stack-manager
+        image: crossplaneio/template-stack-manager:latest
+        args: ["--stack", "stackname"]
+```
+
+A proposed `stack` argument to the TSM would allow the controller to find a matching Stack. The TSM and Stack should reside within the same namespace. The Stack Manager will abide.
+
+### Owner References
+
+The [Controller Reference](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#SetControllerReference) will be set on the TSM `Deployment` so the TSM is removed when the Stack is removed. The TSM can identify the Stack to work on through this controller reference, but this approach may make it difficult for admins to identify which deployments are operating on behalf of Stacks (and their CRDs).
+
+Stacks could also take on a reference to the `Deployment` and `ServiceAccount` assuring that the correct resource will be updated when the Stack is updated. What of the CRDs? Revised CRDs in an updated Stack will be updated because they are unique, but how will Stack CRDs that have been removed be deleted? This is a trickier problem to be considered in future designs.
```

Comment:
> Stacks could also take on a reference to the `Deployment` and `ServiceAccount` assuring that the correct resource will be updated when the Stack is updated.

Deployment only has the name of the stack, so updating would mean just restarting the pod  as stack name is not expected to change?

Reply:
updated in eff1e90a